### PR TITLE
Patch deployment attempt deployment query parameters enum in release

### PR DESCRIPTION
**Pull request by myadav27**
While getting a release below error showing due to mismatch in enum type

image

image

**Solution:** 
Added code in patcher to handle the issue:

### CODE REVIEW

File: vsts-api-patcher/src/main.rs [hunk 1]
```diff
@@ -481,6 +483,260 @@ impl Patcher {
         }
     }
 
+    fn patch_operation_status_in_releases_query_parameters(
+        &mut self,
+        key: &[&str],
+        _value: &JsonValue,
+    ) -> Option<JsonValue> {
+        // Only applies to pipelines specs
```
Comment (johnbatty) [inline vsts-api-patcher/src/main.rs:491]: incorrect comment: should say "release specs"

File: vsts-api-patcher/src/main.rs [hunk 2]
```diff
@@ -481,6 +483,260 @@ impl Patcher {
...
+        if !self.spec_path.ends_with("release.json") {
+            return None;
+        }
+        match key {
+            ["definitions", "DeploymentQueryParameters", "properties", "operationStatus"] => {
```
Comment (johnbatty) [inline vsts-api-patcher/src/main.rs:496]: Looks like this is pretty much the same patch (just with a different description) being applied to two places.

I think we should avoid code duplication and do both of the patches in a single function (e.g. `patch_release_operation_status(...)`), and have that call a single function to return the json with a parameter for the description....

```rust
fn patch_release_operation_status(...) {
    ...
    match key {
            ["definitions", "DeploymentQueryParameters", "properties", "operationStatus"] => {
                Some(patched_operation_status("Query deployment based on deployment operation status.")
            },
            ["definitions", "DeploymentAttempt", "properties", "operationStatus"] => {
                Some(patched_operation_status("Deployment operation status."))
            }
            ...
        }   
```

**Review by johnbatty — CHANGES_REQUESTED** [review]
Thanks - looks good, but I think you can simplify the change.

File: vsts-api-patcher/src/main.rs (hunk 2 above)
Comment (myadav27) [reply vsts-api-patcher/src/main.rs:370]: Updated in the latest commit

File: vsts-api-patcher/src/main.rs (hunk 1 above)
Comment (myadav27) [reply vsts-api-patcher/src/main.rs]: Updated in the latest commit.

**Review by johnbatty — APPROVED** [review]

**Pull request merged**